grafana component creation

khandrew-redhat opened the pull request:
Implements a grafana link component. Within each dropdown in the metrics tab, a grafana icon will appear. The link is construction is done with the response attributes: var-platform, var-cloud_type, var-networkType, var-major_version, and uuid (or id). Once pressed, the link will take you to the respective grafana page.
Integrates Grafana links into the Chaos Dashboard to allow users to directly access related visualizations and metrics from within the dashboard. This enhances observability and streamline navigation between experiment data and monitoring dashboards.

Ticket: https://issues.redhat.com/browse/CHAOS-1193
New Ticket: https://redhat.atlassian.net/browse/CHAOS-1466

 
<img width="1505" height="864" alt="Screenshot 2026-04-02 at 11 44 38 AM" src="https://github.com/user-attachments/assets/d74a08a7-cb2d-4f88-8b0b-4b21de49a744" />
<img width="1501" height="871" alt="Screenshot 2026-04-02 at 11 44 54 AM" src="https://github.com/user-attachments/assets/ecff7778-ccd6-4517-a08a-4916c5cef97c" />

MVarshini commented on src/components/organisms/StorageTable/index.jsx:
```diff
@@ -99,6 +100,14 @@ const StorageTable = () => {
                             <GraphRow doc={doc} />
                           </GridItem>
+                          <GridItem span={12}>
+                            <div style={{ marginTop: "16px", display: "flex", justifyContent: "flex-end" }}>
```
PatternFly generally discourages hard-coded spacing like marginTop: "16px".
Suggest using PF spacing utilities (pf-v5-u-mt-md) or global spacer tokens for consistency and maintainability.

khandrew-redhat commented on src/components/atoms/GrafanaLink/index.jsx:
```diff
@@ -0,0 +1,77 @@
+    if (networkType) {
+      params.append('var-networkType', networkType);
+    }
+    if (nodeCount) {
```
?? operator added so 0 is treated as valid

khandrew-redhat commented on src/components/organisms/StorageTable/index.jsx:
```diff
@@ -99,6 +100,14 @@ const StorageTable = () => {
                           <GridItem span={6}>
                             <GraphRow doc={doc} />
                           </GridItem>
+                          <GridItem span={12}>
```
Working on reformatting content for smaller resolutions

khandrew-redhat committed:
Add Grafana dashboard discovery and storage metrics links

- Fetch dashboard list from Grafana via server (API search + /dashboards fallback)
- Match dashboards to scenario_type; default Grafana base URL on connect form
- Grafana link icon with template vars; remove unused ES dashboard_id fields
- Redux connection state for grafanaBaseUrl and grafanaDashboardIndex

Signed-off-by: Khephren Andrews <khandrew@redhat.com>
Made-with: Cursor
Signed-off-by: Khephren Andrews <khandrew@redhat.com>

base url change

Signed-off-by: Khephren Andrews <khandrew@redhat.com>

paigerube14 reviewed (approved):
if we can squash your commits, everything else looks good to me
